atomics/include/desul/atomics/Compare_Exchange_SYCL.hpp

```diff
@@ -78,16 +82,77 @@ std::enable_if_t<sizeof(T) == 8, T> device_atomic_exchange(T* const dest,
 template <class T, class MemoryOrder, class MemoryScope>
 std::enable_if_t<(sizeof(T) != 8) && (sizeof(T) != 4), T>
 device_atomic_compare_exchange(
-    T* const /*dest*/, T compare, T /*value*/, MemoryOrder, MemoryScope) {
-  assert(false);  // FIXME_SYCL not implemented
+    T* const dest, T compare, T value, MemoryOrder, MemoryScope) {
+  // FIXME_SYCL Use SYCL_EXT_ONEAPI_DEVICE_GLOBAL when available instead
+#ifdef DESUL_SYCL_DEVICE_GLOBAL_SUPPORTED
```

Comment:
Does this actually need to be guarded?  I feel like we could always define it and only guard in the definitions of `[un]lock_address_sycl `